src/vm-vcpu-ref/src/x86_64/interrupts.rs

```diff
@@ -7,36 +7,22 @@
 #![cfg(target_arch = "x86_64")]
 use kvm_bindings::kvm_lapic_state;
 
-// Yanked from byte_order
-macro_rules! generate_read_fn {
-    ($fn_name: ident, $data_type: ty, $byte_type: ty, $type_size: expr, $endian_type: ident) => {
-        fn $fn_name(input: &[$byte_type]) -> $data_type {
-            assert!($type_size == std::mem::size_of::<$data_type>());
-            let mut array = [0u8; $type_size];
-            for (byte, read) in array.iter_mut().zip(input.iter().cloned()) {
-                *byte = read as u8;
-            }
-            <$data_type>::$endian_type(array)
-        }
-    };
+// Helper function that writes the input (which must have 4 bytes) into an i32.
+fn read_le_i32(input: &[i8]) -> i32 {
```

Comment:
Should we write some tests for these ones?
I think we could also keep the assert and check that input length = 4 here.